bot/cogs/doc.py

```diff
@@ -0,0 +1,207 @@
+import functools
+import logging
+import re
+import sys
+from collections import OrderedDict
+from typing import Optional, Tuple
+
+import aiohttp
+import discord
+from bs4 import BeautifulSoup
+from discord.ext import commands
+from markdownify import MarkdownConverter
+from sphinx.ext import intersphinx
+
+log = logging.getLogger(__name__)
+
+
+BASE_URLS = {
+    'aiohttp': "https://aiohttp.readthedocs.io/en/stable/",
+    'discord': "https://discordpy.readthedocs.io/en/rewrite/",
+    'django': "https://docs.djangoproject.com/en/dev/",
+    'stdlib': "https://docs.python.org/%d.%d/" % sys.version_info[:2]
+}
+
+INTERSPHINX_INVENTORIES = {
+    'aiohttp': "https://aiohttp.readthedocs.io/en/stable/objects.inv",
+    'discord': "https://discordpy.readthedocs.io/en/rewrite/objects.inv",
+    'django': "https://docs.djangoproject.com/en/dev/_objects/",
+    'stdlib': "https://docs.python.org/%d.%d/objects.inv" % sys.version_info[:2]
```

Comment:
Keep this static (currently 3.7)

Reply:
hmm, why though? wouldn't it be preferable to link to the most recent stable version (which I assume the bot to be running on)?

Reply:
we'd have to keep changing it for it to be possible to refer to upcoming features. I prefer Volcyys approach to this tbh, because that never needs to be manually maintained.

Reply:
In that case I do recommend using `str.format` or f-strings instead of %-formatting: 

```py
    'stdlib': f"https://docs.python.org/{'.'.join(sys.version_info[:2]}/objects.inv"
```
As %-formatting in later versions of python is heavily frowned upon

Reply:
This I _do_ agree with.